### PR TITLE
added documentation for stat, lstat and exists?

### PR DESCRIPTION
**Pull request by vjdhama**
Added few missing documentations in `file.cr`.

- stat
- lstat
- exists?

### CODE REVIEW

File: src/file.cr [hunk 1]
```diff
@@ -129,6 +129,14 @@ class File < IO::FileDescriptor
     seek value
   end
 
+  # Returns a File::Stat object for the named file or raises 
+  # an error based on LibC.stat(path, stat) return value (See File::Stat)
```
Comment (jhass) [inline src/file.cr]: I don't think that it calls `LibC.stat` under the hood is relevant to the end user (or shouldn't be normally at least). In any case its an implementation detail. How about some generic "or raises `Errno` in the case of an error"?

Comment (vjdhama) [inline src/file.cr]: Sure thing. I'll add the changes.

**Comment by jhass** [comment]
Please squash into a single commit

```sh
git remote add upstream https://github.com/manastech/crystal.git
git fetch upstream
git checkout master
git rebase -i upstream/master
# Leave the first commit at pick, choose squash or fix for the others.
# If there are any conflicts git will inform you about that, resolve them for
# example with git mergetool or manually and with git add, then run
# git rebase --continue, again  watch the output since that might happen for
# each commit.
git log # Look at the updated commit history
git status # should report you're back on your branch
# We need to force push since a rebase edits the commits, effectively
# creating new ones. This is okay to do in a feature branch, the pull
# request will update.
git push -f origin master # Assuming you cloned from your fork
```

Hint: You can also edit your last commit with `git commit --amend`, since that also effectively creates a new commit, you to need to force push afterwards and again you should only do it in feature branches.

Also please consider creating topic branches in the future so you can have multiple PRs open at the same time ;)

**Comment by vjdhama** [comment]
@jhass can you review the updated PR? I added the squashed commit.

**Comment by jhass** [comment]
Thanks!

File: src/file.cr [hunk 2]
```diff
@@ -136,6 +144,14 @@ class File < IO::FileDescriptor
     Stat.new(stat)
   end
 
+  # Returns a `File::Stat` object for the named file or raises 
+  # `Errno` in case of an error.
+  #
+  # ```
+  # echo "foo" > foo
+  # File.stat("foo").size    #=> 4
+  # File.stat("foo").mtime   #=> 2015-09-23 06:24:19 UTC
```
Comment (asterite) [inline src/file.cr]: These two lines should probably say `lstat` :blush:

Comment (jhass) [inline src/file.cr]: Sigh, right. I'll fix it.

**Pull request merged**